arangod/Aql/Query.cpp

```diff
@@ -793,15 +792,13 @@ ExecutionState Query::executeV8(v8::Isolate* isolate, QueryRegistry* registry,
         ExecContext const& exec = ExecContext::current();
 
         // got a result from the query cache
-        if (!exec.isSuperuser()) {
```

Comment:
If the user is a superuser then the user should have access to the collection. Therefore removing this short-cut as Superuser is not well-defined.